### PR TITLE
Poetry File Setup

### PR DESCRIPTION
**Pull request by laurennevill**
# Poetry File Setup
## Description
This pull requests defines the poetry dependencies and the tasks for the project. It also includes a brief description of gatorgrade as well as all of its authors. 
### Type of change

- [x]  Feature
- [ ] Bug Fix
- [ ] Documentation

## Contributors
- @laurennevill

### CODE REVIEW

**Review by burgess01 — APPROVED** [review]
This looks great Lauren! I think including black/flake/etc is a good idea for quality assurance.

File: pyproject.toml
Comment (mariakimheinert) [inline pyproject.toml:28]: Is this the same functionality as the `lint` task? The `lint` task is used in GitHub Actions (see `.github/workflows/main.yml` in PR #28), so if the `pylint` task duplicates the functionality, then the `lint` task should probably be kept instead of `pylint`.

File: pyproject.toml
Comment (mariakimheinert) [inline pyproject.toml:19]: Is there a specific motivation for having these additional linters? I wonder if it may be confusing for others since only `pylint` is required for this project.

File: pyproject.toml
Comment (mariakimheinert) [inline pyproject.toml:23]: Should `package/` be `gatorgrade` in light of the structure that was proposed in the #sep channel?

**Review by mariakimheinert — COMMENTED** [review]
Hello @laurennevill!  Thanks for adding these changes. I have a couple of questions for you below.

File: pyproject.toml
Comment (laurennevill) [reply pyproject.toml:19]: As a group we discussed how adding these additional linters will eventually lead to a cleaner and higher quality final product. We also discussed how some of these enforce a higher level of documentation which could help in the future when thinking about the maintainability and/or sustainability of the system. We did address how adding these extra linters might be more tedious but we overall decided we think their benefits would outweigh the cons.

File: pyproject.toml
Comment (laurennevill) [reply pyproject.toml:28]: Yes that would make more sense, thank you.

File: pyproject.toml
Comment (laurennevill) [reply pyproject.toml:23]: Yes, I will change it to reflect that. Thank you for bringing it to light

File: pyproject.toml
Comment (noorbuchi) [reply pyproject.toml:19]: I'm in favor of including all the other linters. I think most people are familiar with them and have used them before in previous classes. I also think that they can set important requirements for the formatting of the project. Pydocsryle can be especially helpful since there are efforts to generate a website for the tool which requires docstrings to be included. However, I'm interested to hear about what others think about including all these linters.

File: pyproject.toml
Comment (noorbuchi) [reply pyproject.toml]: I think the goal is to include all linters in one task to make running them easier. This would mean that a pylint task is different from a lint task. Is there a specific preference on how to set this up?

Comment (noorbuchi) [reply pyproject.toml]: @Michionlion Yes, your approach is what I was referring to as well. I just wasn't sure how to describe it.  I've set it up like that for multiple projects and haven't had any problems on Mac or Linux. But I'm not sure if it could cause problems on windows either.

File: pyproject.toml
Comment (noorbuchi) [inline pyproject.toml:22]: `package` should be replaced with `gatorgrade`

**Review by noorbuchi — CHANGES_REQUESTED** [review]

**Review by ullrichd21 — CHANGES_REQUESTED** [review]
I left a few things I think should be changed but they're minor!

File: pyproject.toml
Comment (noorbuchi) [reply pyproject.toml:17]: It's fine if you choose to do that, just keep in mind that when you introduce other linters later on, it's possible that you'll need to fix alot at once. On the other hand, if you start off with multiple linters and maintain their passing status, then you'll be confident that the code follows some important standards. Either way, it's up for the class to make that decision.

File: pyproject.toml
Comment (laurennevill) [reply pyproject.toml:18]: Okay thanks for the feedback I will remove it for now until we can expand on the topic in the future!

**Comment by mariakimheinert** [comment]
@laurennevill @ullrichd21 @watson02 @burgess01, please remember that [all pull requests that contain source code contributions must also be reviewed by an industry professional](https://mariakimheinert.com/software-engineering/project/deliverables/individual-contributions/#pull-requests). Pull requests should not be merged until they are approved by all required reviewers. While it is fine that this pull request was merged, please make sure to wait for approval from all required reviewers for future pull requests.

**Comment by noorbuchi** [comment]
@mariakimheinert Since this PR merged into the development branch and is expected to be a part of a larger pull request that merges `develop` into `main`, we thought it would be ok to go ahead and merge this to make some progress in the larger develop branch. #38 contains the changes in this PR in addition to other ones that create the CI configurations.

**Comment by mariakimheinert** [comment]
@noorbuchi, gotcha, I forgot about this. Thanks for the clarification 👍

**Pull request merged**
